### PR TITLE
Add Integral Ethereum and Arbitrum to dex trades

### PR DESCRIPTION
**Pull request by IntegralHQ**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by IntegralHQ** [comment]
@Hosuke , I see you are making some naming changes. For the naming convention can we remove all the "size" references?

integral_arbitrum_size_trades -> integral_arbitrum_trades

We'd like to just call the system Integral. Originally we were going to namespace it as Integral Size, but we think it makes sense to just call it Integral now.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @IntegralHQ 

(I am trying to fix the [dbt seed error](https://github.com/duneanalytics/spellbook/actions/runs/4381438103/jobs/7693061903#step:8:30) and will merge the PR after fixed)

File: models/prices/ethereum/prices_ethereum_tokens.sql
Comment (jeff-dude) [inline models/prices/ethereum/prices_ethereum_tokens.sql:200]: it appears this token is not available on coinpaprika:
https://api.coinpaprika.com/v1/coins/itgr-integral

two options:
- remove from this PR, let new dex tables build without pricing info on this token (the opposite token in a trade could still compute usd amount, based on coalesce used)
- look to add on coinpaprika (https://coinpaprika.com/add/), wait for it to be added before merging this entire PR

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: models/prices/ethereum/prices_ethereum_tokens.sql
Comment (Hosuke) [reply models/prices/ethereum/prices_ethereum_tokens.sql:200]: Hi @IntegralHQ, do you think the correct token-id is `itgr-integral1132`?
https://api.coinpaprika.com/v1/coins/itgr-integral1132
But the token address listed in the API is unmatched to the address in the file.
https://etherscan.io/token/0xd502f487e1841fdc805130e13eae80c61186bc98

File: models/prices/ethereum/prices_ethereum_tokens.sql
Comment (Hosuke) [inline models/prices/ethereum/prices_ethereum_tokens.sql:201]: API: **https://api.coinpaprika.com/v1/coins/itgr-integral1132**

Comment (Hosuke) [reply models/prices/ethereum/prices_ethereum_tokens.sql:201]: SCAN: https://etherscan.io/token/0xd502f487e1841fdc805130e13eae80c61186bc98

**Comment by jeff-dude** [comment]
> The test failure is unrelated to this PR: https://dune.com/queries/2279717?d=1
> 
> (Not sure about the cause yet)

true, unrelated, but would be nice to see resolved in order to get this model running fully end-to-end. looks like pancakeswap on bnb is the problem, which means likely an issue in prod as well

**Comment by Hosuke** [comment]
> > The test failure is unrelated to this PR: https://dune.com/queries/2279717?d=1
> > (Not sure about the cause yet)
> 
> true, unrelated, but would be nice to see resolved in order to get this model running fully end-to-end. looks like pancakeswap on bnb is the problem, which means likely an issue in prod as well

I have tried to debug for a while but have not figured out the exact cause yet, will keep finding.

**Comment by jeff-dude** [comment]
> > > The test failure is unrelated to this PR: https://dune.com/queries/2279717?d=1
> > > (Not sure about the cause yet)
> > 
> > 
> > true, unrelated, but would be nice to see resolved in order to get this model running fully end-to-end. looks like pancakeswap on bnb is the problem, which means likely an issue in prod as well
> 
> Hi @jeff-dude, found the root cause of the issue: At the specific `minute` given, one token in `prices.usd` is having TWO records: https://dune.com/queries/2290413
> 
> minute	blockchain	contract_address	decimals	symbol	price
> 2023-03-18 21:42	bnb	0xce186ad6430e2fe494a22c9edbd4c68794a28b35	18	LOOP	0.054834
> 2023-03-18 21:42	bnb	0xce186ad6430e2fe494a22c9edbd4c68794a28b35	18	LOOP	0.055032
> And hence caused the dupes in the pancakeswap.trades: https://dune.com/queries/2288902
> 
> If this is an expected behaviour from the API, we may need to strengthen the left join conditioning, otherwise we may need to fix the `prices.usd`.
> 
> All dupes in `prices.usd` table given specific `blockchain`, `minute`, `contract_address`: https://dune.com/queries/2290578

this is not expected. thank you for the research and findings!
@aalan3 can you help debug this `prices.usd` bug?

**Comment by Hosuke** [comment]
I have record the `prices.usd` bug in https://github.com/duneanalytics/spellbook/issues/3107
This PR may be unblocked and ready-for-merging (as it has also fixed the token-id issue).
cc. @masquot

**Comment by IntegralHQ** [comment]
Hey @jeff-dude , thanks for working on our pr.

Do you think it would be possible if you can work on this bug in a separate issue and merge our PR soon? Our community has been asking for this listing for quite a while as we quote dune’s data frequently in our community call, so it would be a bit inconvenient that dune does not have our data.

Really appreciate if you can help. Thanks!

**Comment by jeff-dude** [comment]
> Hey @jeff-dude , thanks for working on our pr.
> 
> Do you think it would be possible if you can work on this bug in a separate issue and merge our PR soon? Our community has been asking for this listing for quite a while as we quote dune’s data frequently in our community call, so it would be a bit inconvenient that dune does not have our data.
> 
> Really appreciate if you can help. Thanks!

let me push a temp change to remove the failing test, that way we can see this run fully and feel confident

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
> > Hey @jeff-dude , thanks for working on our pr.
> > Do you think it would be possible if you can work on this bug in a separate issue and merge our PR soon? Our community has been asking for this listing for quite a while as we quote dune’s data frequently in our community call, so it would be a bit inconvenient that dune does not have our data.
> > Really appreciate if you can help. Thanks!
> 
> let me push a temp change to remove the failing test, that way we can see this run fully and feel confident

looked good with removing the unrelated failing test -- reverted and merged 👍 thank you for the patience and contribution!

**Pull request merged**
